leaderboard/make_leaderboard.py

```diff
@@ -0,0 +1,61 @@
+from collections import defaultdict
+
+import click
+from kaggle_environments import make, evaluate
+
+from agents import REGISTERED_AGENTS
+from trueskill_ranking import compute_leaderboard
+
+
+@click.command()
+@click.option('--sigma_threshold', default=1.)
+def make_leaderboard(sigma_threshold):
+    """Compute the leaderboard with TrueSkill rating and create the leaderboard in README.md"""
+    stats_agents = defaultdict(lambda: {"win": 0, "lose": 0, "draw": 0})
+
+    def evaluate_agents(agent_a, agent_b):
+        """Evaluate a vs b and b vs a"""
+        agent_a_func, agent_b_func = REGISTERED_AGENTS[agent_a].func, REGISTERED_AGENTS[agent_b].func
+        a_vs_b = evaluate("connectx", [agent_a_func, agent_b_func], num_episodes=1)
+        b_vs_a = evaluate("connectx", [agent_b_func, agent_a_func], num_episodes=1)
+        score_a = a_vs_b[0][0] + b_vs_a[0][1]
+
+        if score_a > 0:  # agent_a win 1 or 2 games
+            stats_agents[agent_a]["win"] += 1
+            stats_agents[agent_b]["lose"] += 1
+            return 1
+        elif score_a < 0:  # agent_b win 1 or 2 games
+            stats_agents[agent_b]["win"] += 1
+            stats_agents[agent_a]["lose"] += 1
+            return -1
+        else:
+            stats_agents[agent_a]["draw"] += 1
```

Comment:
This may be a step too far, but if one agent always wins when it is player 1 and always loses when it is player 2, it will always count as a draw (whereas that might just be a bad agent).

One possible remedy would be to evaluate in a random order, given the sigma precision will mean this gets run multiple times per agent anyway (I think?).

Also, this is satisfied when agent a wins  1 and loses 1, same as if it draws both. Don't know if that's intentional, as per above

Reply:
- How can an agent be "bad" if it always win when player1? It is an average agent at worst.
- Yes, if we evaluate in a random order and we run many many evaluations, each player should have on average about as many games played as a player1 and a player2. But why not reduce the variance if we can without much cost since we don't have an infinity of evaluations? (The expected result should be the same in both cases IMO).
- Yes, it is intentional.

Maybe I am missing something here. If you think so, tell me.

Reply:
Sorry I didn't explain fully. I meant it would be "bad" because one way it could lose in player 2 but win in player 1 is that it may never have been trained to play as player2. To me that's a major flaw in the agent's generality, and is worse (a less intelligent agent) than being average in both positions. There may be other iterations of that that aren't as severe, but lead to an imbalanced outcome in positions 1 and 2. Maybe that's philosophical though.

Happy to overlook it for our purposes if the fix is complicated / randomisation would be too costly. Let me know what you think!